### PR TITLE
Feature/liquidity changes

### PR DESCRIPTION
**Pull request by OgnjenCBS**

### CODE REVIEW

File: src/database/mysql/migrations/1702477525960-create-pairs-periodic-liquidity-change.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  implements MigrationInterface
+{
+  name = 'createPairsPeriodicLiquidityChange1702477525960';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`CREATE TABLE \`pairs_periodic_liquidity_change\` (
+            \`id\` int NOT NULL AUTO_INCREMENT, 
+            \`base_asset_symbol\` varchar(255) NOT NULL, 
+            \`base_asset_name\` varchar(255) NOT NULL, 
+            \`base_asset_id\` varchar(255) NOT NULL,
```
Comment (DanijelCBS) [inline src/database/mysql/migrations/1702477525960-create-pairs-periodic-liquidity-change.ts:13]: Base asset id and token asset id are not necessary.

File: src/database/mysql/migrations/1702477525960-create-pairs-periodic-liquidity-change.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`CREATE TABLE \`pairs_periodic_liquidity_change\` (
+            \`id\` int NOT NULL AUTO_INCREMENT, 
+            \`base_asset_symbol\` varchar(255) NOT NULL, 
+            \`base_asset_name\` varchar(255) NOT NULL, 
+            \`base_asset_id\` varchar(255) NOT NULL, 
+            \`token_asset_name\` varchar(255) NOT NULL, 
+            \`token_asset_symbol\` varchar(255) NOT NULL, 
+            \`token_asset_id\` varchar(255) NOT NULL, 
+            \`liquidity\` int NOT NULL,
```
Comment (DanijelCBS) [inline src/database/mysql/migrations/1702477525960-create-pairs-periodic-liquidity-change.ts:17]: Set it to float

File: src/modules/pairs-liquidity/dto/pair-periodic-liquidity-change.dto.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export interface PairPeriodicLiquidityChangeDto {
+  baseAssetSymbol: string;
+  baseAssetName: string;
+  baseAssetId: string;
```
Comment (DanijelCBS) [inline src/modules/pairs-liquidity/dto/pair-periodic-liquidity-change.dto.ts:4]: Remove IDs

File: src/database/mysql/migrations/1702477525960-create-pairs-periodic-liquidity-change.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+export class createPairsPeriodicLiquidityChange1702477525960
+  implements MigrationInterface
+{
+  name = 'createPairsPeriodicLiquidityChange1702477525960';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`CREATE TABLE \`pairs_periodic_liquidity_change\` (
+            \`id\` int NOT NULL AUTO_INCREMENT, 
+            \`base_asset_symbol\` varchar(255) NOT NULL, 
+            \`base_asset_name\` varchar(255) NOT NULL,
```
Comment (DanijelCBS) [inline src/database/mysql/migrations/1702477525960-create-pairs-periodic-liquidity-change.ts:12]: Remove base asset name and token asset name.

File: src/modules/pairs-liquidity/pairs-liquidity.controller.ts [hunk 5]
```diff
@@ -20,4 +21,15 @@ export class PairsLiquidityController {
   ): Promise<PageDto<PairLiquidityChangeDto>> {
     return this.pairsLiquidityChangesService.find(assetA, assetB, pageOptions);
   }
+
+  @Get('/periodic-difference/:baseSymbol/:tokenSymbol')
```
Comment (DanijelCBS) [inline src/modules/pairs-liquidity/pairs-liquidity.controller.ts:25]: Rename periodic-difference to history

File: src/modules/pairs-liquidity/pairs-liquidity.service.ts [hunk 6]
```diff
@@ -33,4 +45,41 @@ export class PairsLiquidityService {
 
     return new PageDto(this.mapper.toDtos(data), pageMeta);
   }
+
+  public async getPeriodicChanges(
+    baseAssetSy: string,
```
Comment (DanijelCBS) [inline src/modules/pairs-liquidity/pairs-liquidity.service.ts:50]: Rename to full name baseAssetSymbol

File: src/modules/pairs-liquidity/periodic-liquidity-change.repository.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  ) {}
+
+  public savePeriodcLiqChange(
+    pairLiqChange: PairPeriodicLiquidityChangeEntity,
+  ): Promise<PairPeriodicLiquidityChangeEntity> {
+    return this.periodicLiqChangeRepo.save(pairLiqChange);
+  }
+
+  public findPairPeriodicLiqChange(
+    baseAssetSy: string,
```
Comment (DanijelCBS) [inline src/modules/pairs-liquidity/periodic-liquidity-change.repository.ts:20]: Rename to full name

**Review by DanijelCBS — CHANGES_REQUESTED** [review]

**Review by DanijelCBS — APPROVED** [review]

File: src/modules/pairs-liquidity/pairs-liquidity.controller.ts [hunk 8]
```diff
@@ -20,4 +21,15 @@ export class PairsLiquidityController {
   ): Promise<PageDto<PairLiquidityChangeDto>> {
     return this.pairsLiquidityChangesService.find(assetA, assetB, pageOptions);
   }
+
+  @Get('/history/:baseSymbol/:tokenSymbol')
+  public getPeriodicLiquidityChanges(
+    @Param('baseSymbol') baseSymbol: string,
+    @Param('tokenSymbol') tokenSymbol: string,
+  ): Promise<PairPeriodicLiquidityChangeDto[]> {
+    return this.pairsLiquidityChangesService.getPeriodicChanges(
```
Comment (Str4LeGit) [inline src/modules/pairs-liquidity/pairs-liquidity.controller.ts:30]: Do we need pagination here? I guess that response can get pretty big in a short time.

File: src/modules/pairs-liquidity/pairs-liquidity.service.ts [hunk 9]
```diff
@@ -33,4 +45,37 @@ export class PairsLiquidityService {
...
+    const pairs = await this.pairsService.findAll();
+
+    const relevantPairs = pairs.filter((pair) => pair.liquidity > 200);
+
+    relevantPairs.forEach((pair) => {
+      const pairLiqChange = new PairPeriodicLiquidityChangeEntity();
+      pairLiqChange.baseAssetSymbol = pair.baseAsset;
+      pairLiqChange.tokenAssetSymbol = pair.token;
+      pairLiqChange.liquidity = pair.liquidity;
+      pairLiqChange.updatedAt = new Date();
```
Comment (Str4LeGit) [inline src/modules/pairs-liquidity/pairs-liquidity.service.ts:75]: This should ideally be moved to a mapper class.

**Review by Str4LeGit — APPROVED** [review]

**Pull request merged**